src/main/java/org/opentripplanner/middleware/tripmonitor/TrustedCompanion.java

```diff
@@ -85,32 +87,31 @@ private static OtpUser getRelatedUserFromEmail(OtpUser dependentUser, String acc
             .stream()
             .filter(user -> user.acceptKey.equalsIgnoreCase(acceptKey))
             .findFirst();
-        return relatedUser.map(user -> Persistence.otpUsers.getOneFiltered(eq("email", user.email))).orElse(null);
+        return relatedUser.map(user -> Persistence.otpUsers.getOneFiltered(eq(EMAIL_FIELD_NAME, user.email))).orElse(null);
     }
 
     /**
      * Extract the accept key from the request parameters.
      */
-    private static String getAcceptKeyFromRequest(Request request) {
-        String acceptKey = HttpUtils.getQueryParamFromRequest(request, ACCEPT_KEY, false);
-        if (acceptKey.isEmpty()) {
-            logMessageAndHalt(request, HttpStatus.BAD_REQUEST_400, "Accept key not provided.");
-            return null;
+    private static String getAcceptKeyFromRequest(Request request) throws IllegalArgumentException {
+        // Note: optional is true so a missing accept key will be handled here.
+        String acceptKey = HttpUtils.getQueryParamFromRequest(request, ACCEPT_KEY, true);
+        if (acceptKey == null || acceptKey.isEmpty()) {
+            throw new IllegalArgumentException("Accept key not provided.");
         }
         return acceptKey;
     }
 
     /**
      * Retrieve the dependent user matching the accept key.
      */
-    private static OtpUser getUserFromAcceptKey(Request request, String acceptKey) {
+    private static OtpUser getUserFromAcceptKey(String acceptKey) throws IllegalArgumentException {
         if (acceptKey == null) {
             return null;
         }
         OtpUser user = getUserForAcceptKey(acceptKey);
         if (user == null) {
-            logMessageAndHalt(request, HttpStatus.BAD_REQUEST_400, "Otp user unknown.");
-            return null;
+            throw new IllegalArgumentException("Otp user unknown.");
```

Comment:
OTP in all caps.

Reply:
Updated: https://github.com/ibi-group/otp-middleware/pull/256/commits/99341cb2a25e6633d69fb424482c7936fb6adccc